### PR TITLE
Account Navigation Items

### PR DESCRIPTION
**Pull request by Cdok**
(brief description / PR title)

 **JIRA**: https://mobify.atlassian.net/browse/WEB-1449
 **Linked PRs**: https://github.com/mobify/progressive-web-sdk/pull/743 (**We need a SDK release**)

## Changes
- re-order menu items to have account links at the bottom

## Todos
- [ ] ~(if applicable) analytics events instrumented to measure impact of change~

## How to test-drive this PR
- checkout `WEB-1449__navigation` in the sdk
- run `npm link` (in SDK)
- run `npm install` (is SDK)
- checkout this branch 
- run `npm link progressive-web-sdk`
- run `npm run dev`
- [Preview](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Open menu and check the links

### CODE REVIEW

File: web/app/components/nav-item/index.jsx [hunk 1]
```diff
@@ -24,31 +24,43 @@ NavItemIcon.propTypes = {
 /**
  * Project-specific NavItem which displays an icon on the left.
  */
-export const NavItemWithIcon = (props) => (
-    <NavItem {...props} className="pw--with-icon" />
-)
+export const NavItemWithIcon = (props) => {
+
+    return (
```
Comment (kevinxh) [inline web/app/components/nav-item/index.jsx]: this return is not needed

```jsx
export const NavItemWithIcon = (props) => (
    <NavItem {...props} className={`${props.className} pw--with-icon`} />
)

File: web/app/connectors/_merlins-connector/navigation/parser.js [hunk 2]
```diff
@@ -32,11 +30,42 @@ export const parseNavigation = ($, $content, isLoggedIn) => {
         root.children.push({
             title: $link.text().trim(),
             path: $link.attr('href'),
-            isCategoryLink: true
+            isCategoryLink: true,
+            icon: 'lock'
```
Comment (kevinxh) [inline web/app/connectors/_merlins-connector/navigation/parser.js]: According to the code below, should this be:

```jsx
options: {
   icon: 'lock'
}
```

File: web/app/modals/navigation/constants.js [hunk 3]
```diff
@@ -5,16 +5,19 @@
 export const SIGN_IN_LINK_TEXT = 'Sign In'
 export const SIGN_OUT_LINK_TEXT = 'Sign Out'
 
-export const SIGNED_IN_NAV_ITEM_TYPE = 'AccountLogoutNavItem'
-export const GUEST_NAV_ITEM_TYPE = 'AccountNavItem'
+
+
+export const ACCOUNT_LINK = 'AccountNavItem'
```
Comment (kevinxh) [inline web/app/modals/navigation/constants.js]: `ACCOUNT_LINK` makes me to think of a URL like `/account/...`

Will `ACCOUNT_NAV_ITEM` be better?

**Comment by kevinxh** [comment]
Tested on Merlins and SFCC, works as expected. 👍 

Found two non-blocking inconsistencies, not sure if they are in the scope of this PR, just something noted:
1. Icon and arrows does not have Merlins brand color, whereas the icons in the screenshot in JIRA ticket is colored.

| Actual Screenshot | JIRA Screenshot |
| --- | --- |
| screen shot 2017-07-10 at 3 49 29 pm | image |

2. On Merlins, selected navigation item has highlight style:
screen shot 2017-07-10 at 3 59 07 pm

On SFCC, there is no highlight style for `Gift Certificate` and `Top Seller`... Also, we need to make sure the account nav items can be highlighted..

**Comment by gsaray** [comment]
the chevron issue is fixed, right now SFCC is not getting the selected attribute when an item is selected. I think we can ignore it for now since it's not part of this PR to fix it.

**Pull request merged**